search/search_service/proxy/es_search_proxy.py

```diff
@@ -319,3 +326,150 @@ def search(self, *,
                                                    resource_types=resource_types)
 
         return formatted_response
+
+    def get_document_id_and_field_value_from_key(self,
```

Comment:
just `get_document_json_by_key` would correctly encapsulate the re-usable part of this function. As for the id and field value, that's better done with another function, maybe called `get_field_from_document_json` or just a .get if the output of the first func is simple